vars/dockerExecute.groovy

```diff
@@ -84,15 +116,39 @@ private getDockerOptions(Map dockerEnvVars, Map dockerVolumeBind, def dockerOpti
         }
     }
 
-    if (dockerOptions instanceof CharSequence) {
-        options.add(dockerOptions.toString())
-    } else if (dockerOptions instanceof List) {
-        for (String option : dockerOptions) {
-            options.add "${option}"
+    if (dockerOptions) {
+        if (dockerOptions instanceof CharSequence) {
+            options.add(dockerOptions.toString())
+        } else if (dockerOptions instanceof List) {
+            for (String option : dockerOptions) {
+                options.add "${option}"
+            }
+        } else {
+            throw new IllegalArgumentException("Unexpected type for dockerOptions. Expected was either a list or a string. Actual type was: '${dockerOptions.getClass()}'")
         }
-    } else {
-        throw new IllegalArgumentException("Unexpected type for dockerOptions. Expected was either a list or a string. Actual type was: '${dockerOptions.getClass()}'")
     }
-
     return options.join(' ')
 }
+
+@NonCPS
+boolean isContainerDefined(config) {
+    def imageToContainerMap = config?.jenkinsKubernetes?.imageToContainerMap ?: [:]
+    if (imageToContainerMap.containsKey(env.POD_NAME)) {
+        return imageToContainerMap[env.POD_NAME].containsKey(config.dockerImage)
+    }
+    return false
+}
+
+@NonCPS
+def getContainerDefined(config) {
+    def imageToContainerMap = config?.jenkinsKubernetes?.imageToContainerMap
+    return imageToContainerMap[env.POD_NAME].get(config.dockerImage).toLowerCase()
+}
+
+@NonCPS
+boolean isKubernetes() {
+    if (env.ON_K8S == 'true') {
```

Comment:
What about

```
return Boolean.valueOf(env.ON_K8S)
```

OK, 'TrUe' would also be considered as &lt;true&gt; but checking the string and ```return true``` / ```return false``` is much less convenient. Looks like early ABAP code :-).